src/yunohost/app.py

```diff
@@ -156,10 +156,23 @@ def app_info(app, full=False):
 
 
 def _app_upgradable(app_infos):
+    from packaging import version
 
     # Determine upgradability
     # In case there is neither update_time nor install_time, we assume the app can/has to be upgraded
 
+    # Firstly use the version to know if an upgrade is available
+    app_is_in_catalog = bool(app_infos.get("from_catalog"))
+    upgrade_only_if_version_changes = app_infos["manifest"].get('integration', {}).get("upgrade_only_if_version_changes", None) is True
+    installed_version = version.parse(app_infos["version"])
```

Comment:
In fact I was a bit too optimist about the fact that apps do provide a version key ... Many of them do not ... 

```suggestion
    installed_version = version.parse(app_infos.get("version", "0~ynh0"))
```